numpy/core/tests/test_multiarray.py

```diff
@@ -1760,82 +1761,62 @@ def test_sort(self):
                 msg = 'byte-swapped complex sort, dtype={0}'.format(dt)
                 assert_equal(c, arr, msg)
 
-        # test string sorts.
-        s = 'aaaaaaaa'
-        a = np.array([s + chr(i) for i in range(101)])
+    @pytest.mark.parametrize('s, dtype',  [('aaaaaaaa', 'str'),('aaaaaaaa', np.unicode_)])
+    def test_sort_string(self, s, dtype):
+        a = np.array([s + chr(i) for i in range(101)], dtype = dtype)
```

Comment:
I think this fixes the string/bytes issue
```suggestion
    @pytest.mark.parametrize('s, dtype', [np.bytes_, np.unicode_])
    def test_sort_string(self, dtype):
        # np.array will perform the encoding to bytes for us in the bytes test
        a = np.array(['aaaaaaaa' + chr(i) for i in range(101)], dtype=dtype)
```